lib/styleguide.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const _ = require('lodash'),
+  clayUtils = require('clayutils'),
+  files = require('nymag-fs'),
+  path = require('path'),
+  traverse = require('traverse');
+
+/**
+ * Looks for CSS files associated with a component's variation
+ *
+ * @param {String} styleguide - name of the styleguide that is set in the sites config
+ * @returns {string[]} variations
+ */
+function findVariationFiles(styleguide) {
+  const stylePath = `styleguides/${styleguide}/components`,
+    stylesheets = files.getFiles(stylePath),
+    variations = _.filter(stylesheets, function (file) {
+      // checks for underscores which denotes variations and make sure it's a
+      // css file
+      return file.includes('_') && file.includes('.css');
+    });
+
+  return variations;
+}
+
+/**
+* Grabs variations of all components and returns an array organized by components
+*
+* @param {String} styleguide - name of the styleguide that is set in the sites' config
+* @returns {Object} componentVariations - an object that is organized by
+* component and its variations
+*/
+function getVariations(styleguide) {
+  const foundVariations = styleguide ? findVariationFiles(styleguide) : [],
+    componentVariations = {};
+
+  if (foundVariations.length) {
+    _.forEach(foundVariations, function (variant) {
+      let component = variant.split('_')[0],
+        variantName = path.basename(variant, '.css');
+
+      (componentVariations[component] || (componentVariations[component] = [])).push(variantName);
+    });
+  }
+  return componentVariations;
+}
+
+/**
+* Goes through page data and sets a default for components that
+* have variations but don't have any set. Note: this mutates the array!
+*
+* @param {object} pageData
+* @param {object} variations - all components with variations, usually generated by getVariations
+*/
+function setDefaultVariation(pageData, variations) {
+
+  traverse(pageData).forEach(function (val) {
+    const componentName = _.isObject(val) && val._ref ? clayUtils.getComponentName(val._ref) : undefined;
+
+    if (componentName && variations[componentName]) {
+      if (!val.componentVariation) {
+        _.set(val, 'componentVariation', componentName);
```

Comment:
Should this be `_.set(val, 'componentVariation', variations[componentName]` instead?

Reply:
nope! see https://github.com/clay/amphora-html/issues/31

`variations[componentName]` will return ALL variations. We just want to set a default (which is the component's name). The developer is not expected to set the default variation manually. so `amphora-html` does it automatically.

Reply:
What happens if a component has only a default variation? Based on the logic of getVariations and findVariationFiles, the component won't be in the variations object, so this logic here will not fire.

Reply:
Nvm following IRL discussion